internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1738,47 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
+	exposedPorts := make(map[uint16]string)
+
+	if opts.mainContainerPort != nil {
+		exposedPorts[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+	}
+
+	for name, sidecar := range opts.sidecarConfig {
+		if sidecar.Port != nil {
+			port, err := strconv.Atoi(aws.StringValue(sidecar.Port))
+			if err != nil {
+				return err
+			}
+			if _, ok := exposedPorts[uint16(port)]; !ok {
+				exposedPorts[uint16(port)] = name
+			} else {
```

Comment:
same nesting idea here, this could be:
```suggestion
			if _, ok := exposedPorts[uint16(port)]; ok {
				return err
			}
			exposedPorts[uint16(port)] = name
```